frontend/src/lib/utils.tsx

```diff
@@ -16,6 +16,10 @@ import {
     AnyPropertyFilter,
     PropertyFilter,
     CohortType,
+    CohortCriteriaGroupFilter,
```

Comment:
`lib/utils.tsx` seems like it's grown too large over time. `utils` files are great, but as the size grows it's useful to break them up via some boundary of responsibility, eg `cohortUtils.ts`. This is for the same reason why large programs are broken up into clean modules vs a massive `app.tsx` file.

[Empirical research](http://www.catb.org/~esr/writings/taoup/html/ch04s01.html) estimates optimal module sizes at ~800 LOC (not a hard rule, just a general rule of thumb). Other authors such as Bob Martin recommend even smaller, eg ~300-400 LOC.

This is likely out of scope for the PR, but figured I'd mention it.